modmail/extensions/threads.py

```diff
@@ -0,0 +1,459 @@
+        self.relay_channel = await self.bot.fetch_channel(self.bot.config.thread.relay_channel_id)
+
+    @tasks.loop(count=1)
+    async def fetch_necessary_values(self) -> None:
+        """Fetch the audit log permission."""
+        self.relay_channel: discord.TextChannel = await self.bot.fetch_channel(self.relay_channel.id)
+        self.relay_channel.guild = await self.bot.fetch_guild(self.relay_channel.guild.id)
+        me = await self.relay_channel.guild.fetch_member(self.bot.user.id)
+        self.use_audit_logs = USE_AUDIT_LOGS & me.guild_permissions.view_audit_log
+        logger.debug("Fetched relay channel and use_audit_log perms")
+
+    def cog_unload(self) -> None:
+        """Cancel any tasks that may be running on unload."""
+        self.fetch_necessary_values.cancel()
+
+    def get_ticket(self, id: int, /) -> Ticket:
+        """Fetch a ticket from the tickets dict."""
+        try:
+            ticket = self.bot.tickets[id]
+        except KeyError:
+            raise ThreadNotFoundError(f"Could not find thread from id {id}.")
+        else:
+            return ticket
+
+    # the reason we're checking for a user here rather than a member is because of future support for
+    # a designated server to handle threads and a server where the community resides,
+    # so its possible that the user isn't in the server where this command is run.
+    @commands.command()
+    async def contact(self, ctx: Context, recipient: Union[discord.User, discord.Member]) -> None:
+        """
+        Open a new ticket with a provided recipient.
+
+        This will create a new ticket with the recipient, if a ticket does not already exist.
+            If a ticket already exists, a message will be sent in reply with a link to the existing ticket.
+        If the user is not able to be dmed, a message will be sent to the channel.
+        """
+        if recipient.bot:
+            if recipient == self.bot.user:
+                await ctx.send("I'm perfect, you can't open a ticket with me.")
+                return
+            await ctx.send("You can't open a ticket with a bot.")
+            return
+
+        try:
+            ticket = await self.create_ticket(
+                ctx.message, recipient=recipient, check_for_existing_thread=True
+            )
+        except ThreadAlreadyExistsError:
+            thread = self.get_ticket(recipient.id).thread
+            await ctx.send(
+                f"A thread already exists with {recipient.mention} ({recipient.id})."
+                f"You can find it here: <{BASE_JUMP_URL}/{thread.guild.id}/{thread.id}>",
+                allowed_mentions=discord.AllowedMentions(users=False),
+            )
+            return
+
+        if not await check_can_dm_user(recipient):
+            await ticket.thread.send(
+                USER_NOT_ABLE_TO_BE_DMED_MESSAGE.format(
+                    escape_markdown(f"{recipient.name}#{recipient.discriminator}")
+                )
+            )
```

Comment:
Its possible that they won't be in this guild. This can be added later imo.

Reply:
The reason I haven't implemented this is because the bot would only check the cache, so it may not be helpful. Then again, if the cache doesn't have the one guild, that's more problematic....